lib/Parse/Lexer.cpp

```diff
@@ -873,9 +877,15 @@ void Lexer::lexHexNumber() {
       return formToken(tok::integer_literal, TokStart);
     }
     
+    letterAfterDot = !isDigit(*CurPtr);
     while (isHexDigit(*CurPtr) || *CurPtr == '_')
       ++CurPtr;
     if (*CurPtr != 'p' && *CurPtr != 'P') {
+      if (letterAfterDot) {
```

Comment:
The patch LGTM, but please simplify it by eliminating the "letterAfterDot" boolean.  You should be able to change this condition to:

if (isDigit(PtrOnDot[1])) {